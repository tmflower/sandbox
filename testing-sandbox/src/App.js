import './App.css';
import { Routes, Route } from 'react-router-dom';
import { Login } from './Login';
import { Signup } from './Signup';
import { Todos } from './Todos';
import { Navbar } from './Navbar';
import { useState, useEffect } from 'react';
import axios from 'axios';

export function App() {
  const [todos, setTodos] = useState([]);

  useEffect(() => {
      async function getTodos() {
          const result = await axios.get(`http://localhost:3001/todos`);
          setTodos(result.data);
      }
      getTodos();
  }, [todos]);

  return (
    <div className="App">
      <Navbar />
      <Routes>
        <Route path="/" element={<Todos todos={todos}/>}></Route>
        <Route path="/signup" element={<Signup />}></Route>
        <Route path="/login" element={<Login />}></Route>
        <Route path="/todos" element={<Todos todos={todos}/>}></Route>
      </Routes>
    </div>
  );
}
