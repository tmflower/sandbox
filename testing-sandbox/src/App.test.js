import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { App } from './App';

test("renders without crashing", function() {
  render(
  <MemoryRouter>
      <App />
  </MemoryRouter>);
});

test("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
