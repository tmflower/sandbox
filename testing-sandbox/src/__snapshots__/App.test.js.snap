// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav>
      <a
        class=""
        exact="true"
        href="/signup"
      >
        Sign up
      </a>
      <a
        class=""
        exact="true"
        href="/login"
      >
        Log in
      </a>
      <a
        class=""
        exact="true"
        href="/todos"
      >
        Todos
      </a>
    </nav>
    <div>
      <h1>
        Todos
      </h1>
      <h4>
        Add a new task
      </h4>
      <form>
        <label
          for="task"
        >
          Task:
          <input
            id="task"
            name="task"
            type="text"
            value=""
          />
        </label>
        <button>
          Submit
        </button>
      </form>
      <p />
      <p>
        No tasks yet. Add a new task above.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
